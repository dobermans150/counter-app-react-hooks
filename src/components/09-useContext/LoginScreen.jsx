import React, { useContext } from 'react'
import { UserContext } from './userContext';

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    const userData = {
        id: 'use2134',
        name: 'Christian',
        email: 'christianjguevara@mail.com'
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={ () => setUser( userData ) }
            >
                Login
            </button>
        </div>
    )
}
