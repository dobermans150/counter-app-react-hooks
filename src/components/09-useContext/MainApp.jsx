import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './userContext'

export const MainApp = () => {
    /*    const user = {
           id: 'use2134',
           name: 'Christian',
           email: 'christianjguevara@mail.com'
       } */

    const [ user, setUser ] = useState( {} )

    return (

        <UserContext.Provider value={ { user, setUser } } >
            <AppRouter />
        </UserContext.Provider>

    )
}
