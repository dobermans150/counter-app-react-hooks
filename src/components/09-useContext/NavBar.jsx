import React from 'react'
import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">UseContext</Link>

                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink activeClassName="active" className="nav-link " aria-current="page" to="/home">Home</NavLink>
                        <NavLink activeClassName="active" className="nav-link" to="/login">Login</NavLink>
                        <NavLink activeClassName="active" className="nav-link " to="/about">About</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    )
}
