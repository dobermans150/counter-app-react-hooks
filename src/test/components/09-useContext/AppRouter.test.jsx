import React from 'react';
import { mount } from 'enzyme'

import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/userContext';


describe( 'Pruebas en AppRouter', () => {

    const user = {
        id: '1',
        name: 'John',
        email: 'john@example.com'
    }

    const wrapper = mount(
        <UserContext.Provider
            value={ user }
        >
            <AppRouter />
        </UserContext.Provider>

    )

    test( 'Debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot()

    } )


} )
