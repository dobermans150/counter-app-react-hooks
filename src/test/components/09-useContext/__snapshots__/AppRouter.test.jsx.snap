// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-link "
            href="/home"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-link "
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      Home Screen
    </h1>
    <pre
      className="container"
    />
  </div>
</div>
`;
